Dockerfile

```diff
@@ -1,3 +1,48 @@
-FROM scratch
-CMD eksctl
-COPY --from=weaveworks/eksctl-builder:latest /out /
+# Make sure to run the following commands after changes to this file are made:
+# `make update-build-image-manifest && make push-build-image`
+
+# This digest corresponds to golang:1.12.6-alpine3.9 (when Alpine was pointing to 3.9.4)
+FROM golang@sha256:39677a9dd517a8e5d514dff8e36fa46ecc3fb14618b970bfaf3100cb8fab9ba6
+
+# Build-time dependencies
+RUN apk add --no-cache \
+      curl \
+      git \
+      make \
+      bash \
+      gcc \
+      g++ \
+      libsass-dev \
+      musl-dev \
+      && true
+
+WORKDIR /src
+ENV CGO_ENABLED=0
+COPY install-build-deps.sh go.mod go.sum /src/
+
+# Install all go dependencies and remove the go caches in a single step to reduce the image footprint
+# (caches won't be used later on, we overwrite them by volume-mounting)
+RUN ./install-build-deps.sh && \
+    go install github.com/goreleaser/goreleaser && \
+    go build github.com/kubernetes-sigs/aws-iam-authenticator/cmd/aws-iam-authenticator && \
+    rm -rf /root/.cache/go-build /go/pkg/mod
+
+
+
+# Runtime dependencies. Build the root filesystem of the eksctl image at /out
+RUN mkdir -p /out/etc/apk && cp -r /etc/apk/* /out/etc/apk/
+RUN apk add --no-cache --initdb --root /out \
```

Comment:
I realize it's a bit easier to read this way, but installing all the apks in one RUN command would make the image even smaller.

Reply:
Yes, however we don't aim to make this small, we actually optimise for layer cache here, but the resulting image gets squashed into a small number of layers.